pkg/odo/cli/init/init_test.go

```diff
@@ -0,0 +1,451 @@
+package init
+
+import (
+	"testing"
+
+	"github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/devfile/library/pkg/devfile/parser/data"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/redhat-developer/odo/pkg/odo/cli/init/params"
+	"github.com/redhat-developer/odo/pkg/odo/cli/init/registry"
+	"github.com/redhat-developer/odo/pkg/odo/cmdline"
+	"github.com/redhat-developer/odo/pkg/preference"
+	"github.com/redhat-developer/odo/pkg/testingutil/filesystem"
+)
+
+func TestInitOptions_Complete(t *testing.T) {
+	type fields struct {
+		backends func(*gomock.Controller) []params.ParamsBuilder
+	}
+	tests := []struct {
+		name           string
+		fields         fields
+		cmdlineExpects func(*cmdline.MockCmdline)
+		fsysPopulate   func(fsys filesystem.Filesystem)
+		wantErr        bool
+	}{
+		{
+			name: "directory not empty",
+			fsysPopulate: func(fsys filesystem.Filesystem) {
+				_ = fsys.WriteFile(".emptyfile", []byte(""), 0644)
+			},
+			wantErr: true,
+		},
+		{
+			name: "second backend used",
```

Comment:
Which second backend is used? Can you explicitly mention the backend that is used?

Reply:
In this code, we can see that IsAdequate is called in the first backend and returns false, then called in the second backend that returns true.
The result is that the `ParamsBuild` is called on the second backend, not the first one

```
b1.EXPECT().IsAdequate(gomock.Any()).Return(false)
b2.EXPECT().IsAdequate(gomock.Any()).Return(true)
b2.EXPECT().ParamsBuild().Times(1)
```